src/hotspot/share/utilities/hashtable.cpp

```diff
@@ -136,8 +137,34 @@ static int literal_size(WeakHandle v) {
   return literal_size(v.peek());
 }
 
+const double _resize_factor    = 2.0;     // by how much we will resize using current number of entries
+const int _small_table_sizes[] = {107, 1009, 2017, 4049, 5051, 10103, 20201, 40423 } ;
+const int _small_array_size = sizeof(_small_table_sizes)/sizeof(int);
+
+// possible hashmap sizes - odd primes that roughly double in size.
+// To avoid excessive resizing the odd primes from 4801-76831 and
+// 76831-307261 have been removed.
+const int _large_table_sizes[] =  { 4801, 76831, 307261, 614563, 1228891,
+    2457733, 4915219, 9830479, 19660831, 39321619, 78643219 };
+const int _large_array_size = sizeof(_large_table_sizes)/sizeof(int);
+
+// Calculate next "good" hashtable size based on requested count
+template <MEMFLAGS F> int BasicHashtable<F>::calculate_resize(bool use_large_table_sizes) {
+  int requested = (int)(_resize_factor*number_of_entries());
+  const int* primelist = use_large_table_sizes ? _large_table_sizes : _small_table_sizes;
+  int arraysize =  use_large_table_sizes ? _large_array_size  : _small_array_size;
+  int newsize = primelist[0];
+  int index = 0;
+  for (newsize = primelist[index]; index < (arraysize - 1);
+       newsize = primelist[++index]) {
+    if (requested <= newsize) {
+      break;
+    }
+  }
+  return newsize;
```

Comment:
It is existing code, but could be made clearer as part of this PR:
```
  int newsize;
  for (int i=0; i<arraysize; ++i) {
    newsize = primelist[i];
    if (newsize >= requested)
      break;
  }
  return newsize;
```
Additionally, this method could be made `const`, right?

PS: not a review, just a comment in passing

Reply:
Yes, that is a lot simpler and better.  I'd copied that code from another file without changing it that much.